File: 0771-jewels-and-stones/0771-jewels-and-stones.cpp
class Solution {
public:
    int numJewelsInStones(string J, string S) {
        int arr[256]={0};
        int i,c=0;
        for(i=0;i<J.length();i++){
            arr[J[i]]=1;
        }
        for(i=0;i<S.length();i++){
            if(arr[S[i]]){
                c++;
            }
        }
        return c;
    }
};

File: 0771-jewels-and-stones/0771-jewels-and-stones.java
class Solution {
    public int numJewelsInStones(String jewels, String stones) {
        HashSet<Character> J = new HashSet<>();
        for (int i = 0; i < jewels.length(); i++) J.add(jewels.charAt(i)); //input
        int ans = 0;
        for (int i = 0; i < stones.length(); i++) {
            char ch = stones.charAt(i);
            if (J.contains(ch)) ans++;
        }
        return ans;
    }
}

// USING HASHMAP
// class Solution {
//     public int numJewelsInStones(String jewels, String stones) {
//         HashMap<Character, Integer> J = new HashMap<>();    //only K imp.
//         for (int i = 0; i < jewels.length(); i++) J.put(jewels.charAt(i), 0); //input
//         int ans = 0;
//         for (int i = 0; i < stones.length(); i++) {
//             char ch = stones.charAt(i);
//             if (J.containsKey(ch)) ans++;
//         }
//         return ans;
//     }
// }
